Vdat burst combination

RosieBaish opened the pull request.

chris-baish committed:
Updated Burst to integrate loading code

chris-baish committed:
SpotlessApply

chris-baish committed:
Deleted Vdat, now part of Burst.java

tapfm reviewed (dismissed):
Code looks good, will have to look at using jUnit or similar for testing soon

tapfm commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -58,48 +75,43 @@ public int getAverage() {
-    public int getAttenuator1() {
-        // TODO
-        return 0;
+    public ArrayList<Double> getAttenuator1() {
```
Should return List<Double> rather than ArrayList<Double>

tapfm commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -58,48 +75,43 @@ public int getAverage() {
-        // TODO
-        return 0;
+    public ArrayList<Double> getAttenuator1() {
+        return attenuator1;
```
Should return a clone of the list eg. return new ArrayList<>(attenuator1);

tapfm commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -58,48 +75,43 @@ public int getAverage() {
-        // TODO
-        return 0;
+    public ArrayList<Double> getAttenuator2() {
+        return attenuator2;
```
Similar to attenuator1

tapfm commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -58,48 +75,43 @@ public int getAverage() {
-        // TODO
-        return 0;
+    public ArrayList<Integer> getTxAnt() {
+        return TxAnt;
```
Similar to attenuator1

tapfm commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -58,48 +75,43 @@ public int getAverage() {
-        // TODO
-        return 0;
+    public ArrayList<Integer> getRxAnt() {
+        return RxAnt;
```
Similar to attenuator1

tapfm commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -161,34 +168,31 @@ public double getBatteryVoltage() {
     public List<Double> getV() {
-        // TODO
-        return null;
+        return v;
```
Should return a clone of v, not just a reference to it

tapfm commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -161,34 +168,31 @@ public double getBatteryVoltage() {
-        // TODO
-        return null;
+    public List<Integer> getStartInd() {
+        return startInd;
```
Return a clone

tapfm commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -161,34 +168,31 @@ public double getBatteryVoltage() {
-        // TODO
-        return null;
+    public List<Integer> getEndInd() {
+        return endInd;
```
return a clone

tapfm reviewed (changes requested):
Please fix the returning of lists, so that it is a clone and it is return a List<> in all instances rather than a ArrayList<>. It may be worth considering making lots of the variables immutable anyway.

Additionally please remove ComputationEngineTest.java from the repository.

Apart from that the code seems good.

chris-baish committed:
Removed test class

chris-baish committed:
Add test class to .gitignore

chris-baish committed:
Code Revew Changes

enchom commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -423,4 +425,265 @@ public double getLambdac() {
+
+    private String parseString(String mainString, String searchString) {
+        int[] searchInd = strFind(mainString, searchString);
+        int[] searchCR = strFind(mainString.substring(searchInd[0]), "\r\n");
```
Are you sure this is fine and not windows-specific?

RosieBaish replied:
This is an exact copy of the Matlab. I'm on Linux and it works for me.

enchom replied:
Okay, I will take your word for it.

enchom commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -423,4 +425,265 @@ public double getLambdac() {
+    private double temperature1;
+    private double temperature2;
+    private double batteryVoltage;
+    private ArrayList<Double> attenuator1;
```
Replace all ArrayList with List in field definitions.

RosieBaish replied:
Already merged in, but can be added to a future PR

enchom commented on computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/Burst.java:
```diff
@@ -423,4 +425,265 @@ public double getLambdac() {
+        return Double.parseDouble(parseString(mainString, searchString));
+    }
+
+    private ArrayList<Integer> parseIntArray(String mainString, String searchString) {
```
Replace all ArrayList with List in function return types

RosieBaish replied:
For private functions this removes a large amount of casting.

enchom replied:
There shouldn't be any casting at all if everything is a List?

tapfm committed:
Update Burst.java to finish loading file

tapfm committed:
Merge branch 'master' into vdatBurstCombination

tapfm committed:
Update PlotDataGenerator3D.java

Cast received datatype from Date to double

tapfm committed:
Update Burst.java

fix off by one errors in StartInd and EndInd, andalso an off by one by one error in v
